### ISSUE TITLE
npm run release on Fedora

### ISSUE THREAD

**Issue by paulmadore** [Storj/storjshare-gui#653]
Using all the correct directions, the only problem I've run into trying to compile Storj from source in Fedora is that when you get to the npm run release phase, it fails on fakeroot/dpkg-deb instructions given to it. I've never heard of using dpkg in Fedora for package building, but I'm sure it's doable, I just don't know where to begin. I'm sure I'm one of the few people running Fedora and trying to run Storjshare-gui as well, so I'm not requesting a special release or anything, maybe just some pointers to get to that final hurdle, as the other options aren't working for me either. It's saying that it's on test network but I've put it into production mode; apparently according to #571 this does not mean I'm not actually on the network. I had no choice about Fedora because every time Ubuntu pushes out a minor update they manage to put me into a login loop with a GPU I have mining on that machine. It's very frustrating yet Fedora's driver management seems to be working for me so far. So I'd like some pointers on how to patch my storjshare to overcome the fakeroot/dpkg-deb hurdle it's facing -- without going to the trouble of installing a whole Ubuntu VM.

**Comment by littleskunk** [comment]
https://github.com/electron/electron/blob/master/docs/development/build-instructions-linux.md

**Comment by paulmadore** [comment]
@littleskunk the solution is actually even easier than a random user paging through another project's documentation -- 

`sudo dnf install dpkg`

This will install dpkg-deb which npm can use to finish installing the package. I'm really glad I didn't waste a lot of time formulating this question, since you wasted none answering it, and actually your solution did no solve, therefore that close button was wrongly pushed. You'd know it didn't solve the problem if you had replicated your solution -- what makes you think I could even compile and run Storjshare if my npm were not even installed correctly? The issue was that in most cases npm users are on Debian based systems, which have dpkg by default, so there is not a lot of chatter about it when searching that particular problem.

**Comment by littleskunk** [comment]
If you have any question please join us on https://community.storj.io If you open support questions on github I will close them fast to keep everything organized. Sorry nothing personal.

**Comment by paulmadore** [comment]
Alright man, no issues. I did a little post about the full process to creating an RPM that will install on modern Fedora, if you are interested: https://steemit.com/storj/@phmadore/howto-making-storjshare-gui-run-on-fedora

**Comment by WaspCo** [comment]
I followed your instructions @paulmadore , storjshare does install but won't correctly start from gnome: I get a white empty window. If I go to /opt/storjshare/ and start it from CLI, I get this error => Error: Cannot find module 'storjshare-daemon'

**Comment by paulmadore** [comment]
WaspCo, sorry for delay. Can you give me some log output?

**Comment by WaspCo** [comment]
Hello @paulmadore,

Here is what I did exactly:
- git clone https://github.com/Storj/storjshare-gui.git
- npm run release
I then converted the .deb package to .rpm with alien. But I had to modify it:
- dnf install rpmrebuild
- rpmrebuild -pe storjshare.rpm
I then removed those lines conflicting with the filesystem, saved and quit :
- %dir %attr(0755, root, root) "/"
- %dir %attr(0755, root, root) "/usr/share"
- %dir %attr(0755, root, root) "/usr/share/applications"
Then I did
- yum localinstall storjshare.rpm

It installed successfully on Fedora 26. To stop the previous error I did "storjshare daemon" from command line. Then if I start /opt/storjshare from CLI I get a white empty windows and the error "The futex facility returned an unexpected error code".
